[CCIP-Merge] OCR2 plugins directory only [CCIP-2942]

asoliman92 opened the pull request:
This is first part in merging offchain code from ccip repo (https://github.com/smartcontractkit/ccip) into chainlink

Maintaining history across repos for specific direcotires was complicated so we chose to copy the directory right away.

### Reviewing this PR
Please follow the next steps to make sure no changes were done during merging ocr2/plugins/ccip directory. There's no need to have passing tests to merge into the feature branch.

1. Have both ccip and chainlink repos in the same parent directory
2. ccip repo: run `git checkout 44b23a5a32`. This is the cut commit for this directory that we're merging from.
3. chainlink repo: make sure you're using this branch `ocr2-directory-only`
4. If you're running on Mac please install coreutils to be able to run the following script and check the checksum of files. can use homebrew `brew install coreutils`
5.  Run the following script made by @winder from the parent directory that has both repos.

```
#!/usr/bin/env bash

if [[ "$OSTYPE" == "darwin"* ]]; then
    # macOS
    SUM=gmd5sum
else
    # Assume Linux
    SUM=md5sum
fi

pushd ccip/core/services/ocr2/plugins/ccip/ > /dev/null
echo -e "ccip repo commit: \n$(git log --max-count=1 --oneline --decorate)"
find . -type f -exec $SUM {} \;  > ../files.md5
popd > /dev/null
mv ./ccip/core/services/ocr2/plugins/files.md5 ./chainlink/core/services/ocr2/plugins/files.md5
pushd ./chainlink/core/services/ocr2/plugins/ccip/ > /dev/null
echo -e "chainlink repo commit: \n$(git log --max-count=1 --oneline --decorate)"

echo ""

echo "Source files not modified:"
$SUM -c ../files.md5 2>/dev/null | grep OK | wc -l
echo "Source files modified:"
$SUM -c ../files.md5 2>/dev/null | grep -v OK | wc -l

```

### Expected Output
```
Source files not modified:
     183
Source files modified:
       0     
```
--------------------------------

Original authors:

Co-authored-by: Abdelrahman Soliman (Boda) <2677789+asoliman92@users.noreply.github.com>
Co-authored-by: Agustina Aldasoro <agusaldasoro@users.noreply.github.com>
Co-authored-by: Amir Y <83904651+amirylm@users.noreply.github.com>
Co-authored-by: André Vitor de Lima Matos <andre.matos@smartcontract.com>
Co-authored-by: AnieeG <anindita.ghosh@smartcontract.com>
Co-authored-by: Anindita Ghosh <88458927+AnieeG@users.noreply.github.com>
Co-authored-by: Chunkai Yang <matYang@users.noreply.github.com>
Co-authored-by: Connor Stein <connor.stein@mail.mcgill.ca>
Co-authored-by: Evaldas Latoškinas <34982762+elatoskinas@users.noreply.github.com>
Co-authored-by: Jean Arnaud <jarnaud@users.noreply.github.com>
Co-authored-by: Justin Kaseman <justinkaseman@live.com>
Co-authored-by: Makram <makramkd@users.noreply.github.com>
Co-authored-by: Makram Kamaleddine <makramkd@users.noreply.github.com>
Co-authored-by: Mateusz Sekara <mateusz.sekara@gmail.com>
Co-authored-by: Matt Yang <matt.yang@smartcontract.com>
Co-authored-by: Patrick <patrick.huie@smartcontract.com>
Co-authored-by: Rens Rooimans <github@rensrooimans.nl>
Co-authored-by: Roman Kashitsyn <roman.kashitsyn@gmail.com>
Co-authored-by: Roman Kashitsyn <roman.kashitsyn@smartcontract.com>
Co-authored-by: Ryan Stout <ryan.stout@smartcontract.com>
Co-authored-by: Will Winder <wwinder.unh@gmail.com>
Co-authored-by: connorwstein <connor.stein@mail.mcgill.ca>
Co-authored-by: dimitris <dimitrios.kouveris@smartcontract.com>
Co-authored-by: dimkouv <dimitrios.kouveris@smartcontract.com>
Co-authored-by: nogo <0xnogo@gmail.com>
Co-authored-by: nogo <110664798+0xnogo@users.noreply.github.com>
Co-authored-by: valerii-kabisov-cll <172247313+valerii-kabisov-cll@users.noreply.github.com>

winder reviewed (approved):
LGTM

```
ccip repo commit:
44b23a5a32 (HEAD) CCIP-2791 Long term solution for snoozing and caching CommitRoots (#1182)
chainlink repo commit:
1ad9854aa7 (HEAD -> ocr2-directory-only, origin/ocr2-directory-only) Merge remote-tracking branch 'ccipr/ocr2-merge' into ocr2-directory-only

Source files not modified:
183
Source files modified:
0
```

Commit:
Copy over core/services/ocr2/plugins/ccip from ccip repo

This is first part in merging offchain code from ccip repo (https://github.com/smartcontractkit/ccip) into chainlink

Maintaining history across repos for specific direcotires was complicated so we chose to copy the directory right away.

Original authors (anyone who have changes on the directory):
Co-authored-by: Abdelrahman Soliman (Boda) <2677789+asoliman92@users.noreply.github.com>
Co-authored-by: Agustina Aldasoro <agusaldasoro@users.noreply.github.com>
Co-authored-by: Amir Y <83904651+amirylm@users.noreply.github.com>
Co-authored-by: André Vitor de Lima Matos <andre.matos@smartcontract.com>
Co-authored-by: AnieeG <anindita.ghosh@smartcontract.com>
Co-authored-by: Anindita Ghosh <88458927+AnieeG@users.noreply.github.com>
Co-authored-by: Chunkai Yang <matYang@users.noreply.github.com>
Co-authored-by: Connor Stein <connor.stein@mail.mcgill.ca>
Co-authored-by: Evaldas Latoškinas <34982762+elatoskinas@users.noreply.github.com>
Co-authored-by: Jean Arnaud <jarnaud@users.noreply.github.com>
Co-authored-by: Justin Kaseman <justinkaseman@live.com>
Co-authored-by: Makram <makramkd@users.noreply.github.com>
Co-authored-by: Makram Kamaleddine <makramkd@users.noreply.github.com>
Co-authored-by: Mateusz Sekara <mateusz.sekara@gmail.com>
Co-authored-by: Matt Yang <matt.yang@smartcontract.com>
Co-authored-by: Patrick <patrick.huie@smartcontract.com>
Co-authored-by: Rens Rooimans <github@rensrooimans.nl>
Co-authored-by: Roman Kashitsyn <roman.kashitsyn@gmail.com>
Co-authored-by: Roman Kashitsyn <roman.kashitsyn@smartcontract.com>
Co-authored-by: Ryan Stout <ryan.stout@smartcontract.com>
Co-authored-by: Will Winder <wwinder.unh@gmail.com>
Co-authored-by: connorwstein <connor.stein@mail.mcgill.ca>
Co-authored-by: dimitris <dimitrios.kouveris@smartcontract.com>
Co-authored-by: dimkouv <dimitrios.kouveris@smartcontract.com>
Co-authored-by: nogo <0xnogo@gmail.com>
Co-authored-by: nogo <110664798+0xnogo@users.noreply.github.com>
Co-authored-by: valerii-kabisov-cll <172247313+valerii-kabisov-cll@users.noreply.github.com>

0xnogo reviewed (approved):
I got the expected result:
```
ccip repo commit:
44b23a5a32 (HEAD) CCIP-2791 Long term solution for snoozing and caching CommitRoots (#1182)
chainlink repo commit:
3469d502e8 (HEAD -> ocr2-directory-only, origin/ocr2-directory-only) Copy over core/services/ocr2/plugins/ccip from ccip repo

Source files not modified:
     183
Source files modified:
       0
```

winder reviewed (approved):
LGTM
```
ccip repo commit:
44b23a5a32 (HEAD) CCIP-2791 Long term solution for snoozing and caching CommitRoots (#1182)
chainlink repo commit:
3469d502e8 (HEAD -> ocr2-directory-only, origin/ocr2-directory-only) Copy over core/services/ocr2/plugins/ccip from ccip repo

Source files not modified:
183
Source files modified:
0
```
